src/main/java/teammates/common/datatransfer/FeedbackSessionResultsBundle.java

```diff
@@ -1732,4 +1733,22 @@ public boolean hasResponseToInstructorOrGeneral() {
         }
         return false;
     }
+
+    /**
+     * Returns list of responses with identities of giver/recipients NOT hidden which is used for
+     * anonymous result calculation.
+     * @param question question whose responses are required
+     * @return List of responses
+     */
+    public List<FeedbackResponseAttributes> getActualResponses(
+            FeedbackQuestionAttributes question) {
+        String questionId = question.getId();
+        List<FeedbackResponseAttributes> responses;
+        //Get all actual responses for this question
+        responses = actualResponses.stream()
+                       .filter(response -> response.feedbackQuestionId.equals(questionId))
+                       .collect(Collectors.toList());
+        responses.sort(compareByGiverRecipientQuestion);
```

Comment:
Although we don't need sorting, but even if it sorted, it won't make any difference. Then why decouple it?

Reply:
- One obvious reason is we are doing extra work which is useless
- In addition, both the method name and the Javadoc doesn't say it is sorted, which might be error-prone in the further.
- It will complicate your test cases and currently, you are not testing the sorting behavior.